backend/src/main/kotlin/org/loculus/backend/controller/FilesController.kt

```diff
@@ -0,0 +1,47 @@
+package org.loculus.backend.controller
+
+import io.swagger.v3.oas.annotations.Operation
+import io.swagger.v3.oas.annotations.Parameter
+import io.swagger.v3.oas.annotations.security.SecurityRequirement
+import org.loculus.backend.api.FileIdAndUrl
+import org.loculus.backend.auth.AuthenticatedUser
+import org.loculus.backend.auth.HiddenParam
+import org.loculus.backend.service.files.FilesDatabaseService
+import org.loculus.backend.service.files.S3Service
+import org.loculus.backend.service.groupmanagement.GroupManagementPreconditionValidator
+import org.springframework.validation.annotation.Validated
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.bind.annotation.RestController
+
+@RestController
+@RequestMapping("/files")
+@Validated
+@SecurityRequirement(name = "bearerAuth")
+class FilesController(
+    private val filesDatabaseService: FilesDatabaseService,
+    private val s3Service: S3Service,
+    private val groupManagementPreconditionValidator: GroupManagementPreconditionValidator,
+) {
+
+    @Operation(description = "Requests S3 presigned URLs to upload files")
+    @PostMapping("/request-uploads")
```

Comment:
Should we maybe actually do `/request-upload` (not plural)?

Reply:
okay, sure

Reply:
Done: 3a365999e5c047f06ade60c61ae1513dae8b2875